tests/RedisMutexTest.php

```diff
@@ -2,168 +2,129 @@
 namespace yiiunit\extensions\redis;
 
 use Yii;
-use yii\helpers\ArrayHelper;
 use yii\redis\Connection;
 use yii\redis\Mutex;
 
-
 /**
- * Tests for Redis mutex
- *
+ * Class for testing redis mutex
  * @group redis
  * @group mutex
  */
 class RedisMutexTest extends TestCase
 {
-    /**
-     * @param Connection $connection
-     * @param array $params
-     *
-     * @return Mutex
-     * @throws \yii\base\InvalidConfigException
-     */
-    protected function getMutexInstance(Connection $connection, array $params = [])
-    {
-        return Yii::createObject(ArrayHelper::merge($params, [
-            'class' => Mutex::className(),
-            'redis' => $connection,
-        ]));
-    }
+    protected static $mutexName = 'testmutex';
 
-    public function testAcquireLock()
-    {
-        $lockKeyPrefix = 'foo_';
-        $testLockName = 'the_lock';
-
-        $connection = $this->getConnection(true);
-        $mutex = $this->getMutexInstance($connection, [
-            'keyPrefix' => $lockKeyPrefix,
-        ]);
+    protected static $mutexPrefix = 'prefix';
 
-        $this->assertTrue($mutex->acquire($testLockName));
-        $this->assertNotNull($connection->get($lockKeyPrefix . $testLockName));
-    }
+    private static $_keys = [];
 
-    public function testReleaseLock()
+    public function testBasicAcquireAndRelease()
     {
-        $lockKeyPrefix = 'bar_';
-        $testLockName = 'the_lock';
+        $mutex = $this->createMutex();
 
-        $connection = $this->getConnection(true);
-        $mutex = $this->getMutexInstance($connection, [
-            'keyPrefix' => $lockKeyPrefix,
-        ]);
+        $this->assertFalse($mutex->release(static::$mutexName));
+        $this->assertMutexKeyNotInRedis();
+
+        $this->assertTrue($mutex->acquire(static::$mutexName));
+        $this->assertMutexKeyInRedis();
+        $this->assertTrue($mutex->release(static::$mutexName));
+        $this->assertMutexKeyNotInRedis();
 
-        $mutex->acquire($testLockName);
-        $this->assertTrue($mutex->release($testLockName));
-        $this->assertNull($connection->get($lockKeyPrefix . $testLockName));
+        $this->assertFalse($mutex->release(static::$mutexName));
```

Comment:
This test imitates real mutex life scenarios to be carried of. Like no double release, no ownership overtaking. I can, I would and I will split this to parts. I just hope it wouldn't block making this into 2.0.6.